app/modules/code_provider/code_provider_service.py

```
@@ -1,26 +1,52 @@
+            return await self.local_service.get_project_structure_async(
+                project_id, path
+            )
+        elif self.github_service:
+            # GitHub repository
+            return await self.github_service.get_project_structure_async(
+                project_id, path
+            )
+        else:
+            raise Exception(
+                f"GitHub not configured and project {project_id} has no local repo_path"
+            )
```

Ruff (0.14.3)

Create your own exception

(TRY002)